### DISCUSSION TITLE
So are we just done with Docker and no one mentioned it?

### DISCUSSION THREAD

**Discussion by TheGorf** [mastodon/mastodon#24925] [General]
The main documentation page no longer has anything about installing/upgrading via Docker. Almost all info online now about running Mastodon in docker or kubernetes is via third-party sites. I've basically had to nuke and pave my last install because the 4.0 -> 4.1 update was a disaster. And trying to build a from-scratch deployment of 4.1 via Docker seems to be flaky at best. Cloning the repo? Really?  So are we regressing back to the early days of the 2000's where I now have to physically launch server instances and build the code on each one to achieve resiliency and scale? I thought we were trying to achieve some cutting edge approaches to the Fediverse? Someone help me out. What am I missing here? Packaging docker containers is easy and scalable.

**Comment by renchap** [comment]
While the documentation is really lacking on this, there are official containers images for Mastodon, you can find them here: https://github.com/mastodon/mastodon/pkgs/container/mastodon (they are linked in the readme).

We are using them on mastodon.social and mastodon.online, so they are well maintained.

There are per-version tags, as well as a `latest` tag (pointing to the latest release)`, an `edge` tag with the last commit from the `main` branch, and `nightly` releases built every day. 

Note that we dont recommend running anything else than released versions.

Also note that there is an official Helm chart: https://github.com/mastodon/chart
It is not well maintained yet, but we intend to give it some love very soon.
